lib/rsvp/-internal.js

```diff
@@ -204,6 +204,7 @@ export function invokeCallback(settled, promise, callback, detail) {
     if (value === TRY_CATCH_ERROR) {
       failed = true;
       error = value.error;
```

Comment:
now that you've removed `value = null` this makes much more sense :D disregard the comment.